src/nwbuild.js

```diff
@@ -77,41 +80,47 @@ import { log } from "./log.js";
  */
 const nwbuild = async (options) => {
   let nwDir = "";
-  let nwPkg = {};
+  let nwPkg = undefined;
   let cached;
   let built;
   let releaseInfo = {};
   try {
-    //Check if package.json exists in the srcDir
-    nwPkg = JSON.parse(await readFile(`${options.srcDir}/package.json`));
+    let files = [];
+    let patterns = options.srcDir.split(" ");
+
+    for (const pattern of patterns) {
+      let contents = await glob(pattern);
+      for (const content of contents) {
+        if (basename(content) === "package.json" && nwPkg === undefined) {
+          nwPkg = JSON.parse(await readFile(content));
+        }
+        files.push(...contents);
```

Comment:
Shouldn't this go outside the loop?
I'm getting 

```
[ ERROR ] 2023-01-05T08:19:05.751Z Invalid array length
```

with it inside the loop when I have a large number of matched files.

Reply:
Additionally, this seems to conflict:

You describe `srcDir` as: "Directory to hold NW app files unless or array of file glob patterns"

But then you do this:
https://github.com/nwutils/nw-builder/blob/af32f1095578ac35c2cabcc7186ef8b4986277cb/src/nwbuild.js#L89

If a user provides an array of glob patterns (like you document is allowed), it will fail.